pycernan/avro/base_client.py

```diff
@@ -15,7 +17,9 @@ def _rand_u64():
 
 class BaseClient(Client):
     def _send_exact(self, sock, payload):
-        sock.sendall(payload)
+        metrics.bytes_sent.inc(len(payload))
```

Comment:
re: Move metric to after completing task, not before

Reply:
Hmm, it's arguable actually as we -will- attempt to send the data here and we may or may not raise an exception while trying to send and we will never know how many bytes we transfered because we are using sendall() rather than our own loop. If we were to implement our own loop I would agree but in this case I am unsure if there is a meaningful difference.

Reply:
Would it be valuable to implement our own loop here? Should be fairly trivial and would get us more meaningful metrics on what we're sending over the wire, particularly because `send()` returns the number of bytes sent. Otherwise, it might make sense to make this metric `bytes_being_sent` or something more accurate.

Reply:
Yeah I might do. Will send a follow up PR that implements said loop. For now I think it's fine as it only affects failures (which we expect to have close to 0 instances of).